hypershift-operator/controllers/nodepool/aws.go

```diff
@@ -166,6 +155,42 @@ func awsMachineTemplateSpec(infraName string, hostedCluster *hyperv1.HostedClust
 	return awsMachineTemplateSpec, nil
 }
 
+func awsAdditionalTags(nodePool *hyperv1.NodePool, hostedCluster *hyperv1.HostedCluster, infraName string) capiaws.Tags {
+	tags := capiaws.Tags{}
+	for _, tag := range append(nodePool.Spec.Platform.AWS.ResourceTags, hostedCluster.Spec.Platform.AWS.ResourceTags...) {
+		tags[tag.Key] = tag.Value
+	}
+
+	// We enforce the AWS cluster cloud provider tag here.
+	// Otherwise, this would race with the HC defaulting itself hostedCluster.Spec.Platform.AWS.ResourceTags.
+	key := awsClusterCloudProviderTagKey(infraName)
+	if _, ok := tags[key]; !ok {
+		tags[key] = infraLifecycleOwned
+	}
+	return tags
+}
+
+func (c *CAPI) awsMachineTemplate(templateNameGenerator func(spec any) (string, error)) (*capiaws.AWSMachineTemplate, error) {
+	spec, err := awsMachineTemplateSpec(c.capiClusterName, c.hostedCluster, c.nodePool, c.cpoCapabilities.CreateDefaultAWSSecurityGroup, c.releaseImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate AWSMachineTemplateSpec: %w", err)
+	}
+
+	templateName, err := templateNameGenerator(spec)
```

Comment:
is this changing how the name is created? i.e. is it causing a ref change and so a rollout?